nrfs/helpers/dvfs_oppoint.c

```diff
@@ -0,0 +1,105 @@
+	(0UL) /*!< Position of LOCKRANGELOWN field.                                    */
+#define ABB_TRIM_LOCKRANGE_LOCKRANGELOWN_Msk_L (0xFFUL << ABB_TRIM_LOCKRANGE_LOCKRANGELOWN_Pos_L)
+
+#define ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHN_Pos_L                                                    \
+	(8UL) /*!< Position of LOCKRANGEHIGHN field.                                  */
+#define ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHN_Msk_L (0xFFUL << ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHN_Pos_L)
+
+#define ABB_TRIM_LOCKRANGE_LOCKRANGELOWP_Pos_L                                                     \
+	(16UL) /*!< Position of LOCKRANGELOWP field.                                     */
+#define ABB_TRIM_LOCKRANGE_LOCKRANGELOWP_Msk_L (0xFFUL << ABB_TRIM_LOCKRANGE_LOCKRANGELOWP_Pos_L)
+
+#define ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHP_Pos_L                                                    \
+	(24UL) /*!< Position of LOCKRANGEHIGHP field.                                   */
+#define ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHP_Msk_L (0xFFUL << ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHP_Pos_L)
+
+#define ABB_LOCKRANGE(low_n, high_n, low_p, high_p)                                                \
+	(MOVE_AND_MASK_32((low_n),                                                                 \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGELOWN_Msk_L),                                \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGELOWN_Pos_L)) |                              \
+	 MOVE_AND_MASK_32((high_n),                                                                \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHN_Msk_L),                               \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHN_Pos_L)) |                             \
+	 MOVE_AND_MASK_32((low_p),                                                                 \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGELOWP_Msk_L),                                \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGELOWP_Pos_L)) |                              \
+	 MOVE_AND_MASK_32((high_p),                                                                \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHP_Msk_L),                               \
+			  (ABB_TRIM_LOCKRANGE_LOCKRANGEHIGHP_Pos_L)))
+
+#define ABB_RINGO(ringo_target_n, ringo_target_p)                                                  \
+	(MOVE_AND_MASK_32((ringo_target_n),                                                        \
+			  (ABB_TRIM_RINGO_RINGOTARGETVALN_Msk),                                    \
+			  (ABB_TRIM_RINGO_RINGOTARGETVALN_Pos)) |                                  \
+	 MOVE_AND_MASK_32((ringo_target_p),                                                        \
+			  (ABB_TRIM_RINGO_RINGOTARGETVALP_Msk),                                    \
+			  (ABB_TRIM_RINGO_RINGOTARGETVALP_Pos)))
+#else
+#define ABB_LOCKRANGE(low_n, high_n, low_p, high_p) (0)
+#define ABB_RINGO(ringo_target_n, ringo_target_p) (0)
+#endif
+
+static struct dvfs_oppoint_data dvfs_oppoints_data[DVFS_FREQ_COUNT] = {
```

Comment:
Shouldn't this be `const`? Allowing caller to modify the variable sounds really bad to me. The same applies to the function, i.e. it should return pointer to `const` data (if it not desired for callers to modify this). Note that additional benefit of constifying this would be reducing RAM usage.